core/src/test/scala/unit/kafka/admin/ListConsumerGroupTest.scala

```diff
@@ -64,28 +67,177 @@ class ListConsumerGroupTest extends ConsumerGroupCommandTest {
     }, s"Expected to show groups $expectedListing, but found $foundListing")
 
-    val expectedListingStable = Set(
-      new ConsumerGroupListing(group, false, Optional.of(ConsumerGroupState.STABLE)))
+    val expectedListingStable = Set.empty[ConsumerGroupListing]
 
     foundListing = Set.empty[ConsumerGroupListing]
     TestUtils.waitUntilTrue(() => {
-      foundListing = service.listConsumerGroupsWithState(Set(ConsumerGroupState.STABLE)).toSet
+      foundListing = service.listConsumerGroupsWithFilters(Set.empty, Set(ConsumerGroupState.PREPARING_REBALANCE)).toSet
       expectedListingStable == foundListing
     }, s"Expected to show groups $expectedListingStable, but found $foundListing")
   }
 
-  @ParameterizedTest
-  @ValueSource(strings = Array("zk", "kraft"))
-  def testConsumerGroupStatesFromString(quorum: String): Unit = {
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumAndGroupProtocolNames)
+  @MethodSource(Array("getTestQuorumAndGroupProtocolParametersClassicGroupProtocolOnly"))
+  def testListConsumerGroupsWithTypesClassicProtocol(quorum: String, groupProtocol: String): Unit = {
+    val simpleGroup = "simple-group"
+    val protocolGroup = "protocol-group"
+
+    addSimpleGroupExecutor(group = simpleGroup)
+    addConsumerGroupExecutor(numConsumers = 1)
+    addConsumerGroupExecutor(numConsumers = 1, group = protocolGroup, groupProtocol = groupProtocol)
+
+    val cgcArgs = Array("--bootstrap-server", bootstrapServers(), "--list", "--type")
+    val service = getConsumerGroupService(cgcArgs)
+
+    val expectedListingStable = Set.empty[ConsumerGroupListing]
+
+    val expectedListing = Set(
+      new ConsumerGroupListing(
+        simpleGroup,
+        true,
+        Optional.of(ConsumerGroupState.EMPTY),
+        Optional.of(GroupType.CLASSIC)
+      ),
+      new ConsumerGroupListing(
+        group,
+        false,
+        Optional.of(ConsumerGroupState.STABLE),
+        Optional.of(GroupType.CLASSIC)
+      ),
+      new ConsumerGroupListing(
+        protocolGroup,
+        false,
+        Optional.of(ConsumerGroupState.STABLE),
+        Optional.of(GroupType.CLASSIC)
+      )
+    )
+
+    // No filters explicitly mentioned. Expectation is that all groups are returned.
+    var foundListing = Set.empty[ConsumerGroupListing]
+    TestUtils.waitUntilTrue(() => {
+      foundListing = service.listConsumerGroupsWithFilters(Set.empty, Set.empty).toSet
+      expectedListing == foundListing
+    }, s"Expected to show groups $expectedListing, but found $foundListing")
+
+    // When group type is mentioned:
+    // Old Group Coordinator returns empty listings if the type is not Classic.
+    // New Group Coordinator returns groups according to the filter.
+    foundListing = Set.empty[ConsumerGroupListing]
+    TestUtils.waitUntilTrue(() => {
+      foundListing = service.listConsumerGroupsWithFilters(Set(GroupType.CONSUMER), Set.empty).toSet
+      expectedListingStable == foundListing
```

Comment:
nit: Should we remove `expectedListingStable` and directly use `Set.empty[ConsumerGroupListing]` here?